### ISSUE TITLE
False positive `E0308` for macro 2.0 calling another macro in the same crate

### ISSUE THREAD

**Issue by EienMiku** [rust-lang/rust-analyzer#21800]
**rust-analyzer version**: rust-analyzer version: 0.4.2822-standalone (e0269ce3f2 2026-03-10)

**rustc version**: rustc 1.96.0-nightly (e370b60cf 2026-03-07)

**editor or extension**: VS Code 1.112.0; rust-analyzer extension 0.4.2822;

**relevant settings**: N/A

**code snippet to reproduce**:
crate `a`, `src/lib.rs`:
```rust
#![feature(decl_macro)]

pub macro inner() {
    true
}

pub macro outer() {{
    fn __check() -> bool {
        inner!()
    }
    true
}}
```

crate `b`, `src/main.rs`:
```rust
use a::outer;

fn main() {
    outer!();
}
```

**Expected:** No error. `cargo +nightly build` compiles.
**Actual behavior:**
> expected bool, found ()
<img width="915" height="250" alt="Image" src="https://github.com/user-attachments/assets/61f6a136-6ed9-4569-ae26-05ceb977cced" />

**Comment by ChayimFriedman2** [comment]
We don't implement macro 2.0 hygiene.
